app/main/views/service_settings/index.py

```diff
@@ -226,8 +226,10 @@ def request_to_go_live(service_id):
 @user_has_permissions("manage_service")
 @user_is_gov_user
 def submit_request_to_go_live(service_id):
-    ticket_message = render_template("support-tickets/go-live-request.txt") + "\n"
+    if not current_service.go_live_checklist_completed:
```

Comment:
Just a little suggestion so that this matches the exact logic in the HTML

```suggestion
    if (not current_service.go_live_checklist_completed) or (current_service.able_to_accept_agreement and not current_service.organisation.agreement_signed):
```

Reply:
ok, need to mock these two conditions out as well now

Reply:
@klssmith services in self-approving orgs won't have the agreement signed (seeing that in failed tests)

Reply:
The requirements for going live are the same for "normal" vs self-approving orgs, so they should still need the agreement signed, it looks like the tests weren't bothering to mock the agreement being signed since we weren't checking it in the endpoint before now.

For the `test_request_to_go_live_is_sent_to_organiation_if_can_be_approved_by_organisation` test, it passes if you add this mock that we use elsewhere:
```py
mocker.patch(
    "app.models.organisation.Organisation.agreement_signed",
    new_callable=PropertyMock,
    return_value=True,
    create=True,
)
```

The `test_request_to_go_live_displays_go_live_notes_in_zendesk_ticket` test should also be changed to have the agreement signed. These changes fix it

```py
# In the second mock
             "Org 1",
+            agreement_signed=True,
             request_to_go_live_notes="This service is not allowed to go live",
```

```py
         "Organisation type: Central government\n"
-        "Agreement signed: No (organisation is Org 1, a crown body). {expected_go_live_notes}\n"
+        "Agreement signed: Yes, for Org 1. {expected_go_live_notes}\n"
         "\n"
```

Reply:
thanksfor this. yeah wasn't mocking before, but had failed tests locally when trying to mock.